tasks/go.py

```diff
@@ -373,7 +373,7 @@ def check_mod_tidy(ctx, test_folder="testmodule"):
         errors_found = []
         for mod in DEFAULT_MODULES.values():
             with ctx.cd(mod.full_path()):
-                ctx.run("go mod tidy -compat=1.17")
```

Comment:
What's the link with bumping Go to `1.20.8` ?

Reply:
None, but I asked @KSerrania if we wanted to keep that (seemed weird to me to have a 1.17 hardcoded) and the answer was to remove it

Reply:
Ahahah @KSerrania approval is divine, hence LGTM !